js/analyzescorer.js

```diff
@@ -47,55 +52,74 @@ AnalyzeScorer.prototype.runQueue = function() {
 };
 
 /**
- * scoring function that returns results based on the resultobj from the analyzer matched with
- * the scorearrays in the scoring config.
- * @param obj
+ * Looks up the score based on the scorename in the object and calls calculate score
+ * if a scoreObject is found.
+ * @param {object} obj The resultobject from the resultarray.
  * @returns {{name: (analyzerScoring.scoreName), score: number, text: string}}
  */
 AnalyzeScorer.prototype.genericScore = function( obj ) {
-	if ( typeof obj !== "undefined" ) {
+	if ( !isUndefined( obj ) ) {
```

Comment:
If you would inverse this if and use an early return, you could remove one level of nesting and get rid of the last if statement.